database_normalization/database_normalization.md

```diff
@@ -0,0 +1,370 @@
+
+We tend to only include in these entity tables the data elements that have a single value that is relatively stable (like medication generic name, device bar code, patient date of birth, or medical procedure billing code).  We may exclude things from entity tables that could have multiple values or change frequently (for example, we would probably exclude patient insurance provider from the patient table).
+
+**Interactions: How do Entities Interact?**
+
+Then, we make tables that represent **interactions** between entities.  For example, interactions between patients and medications might be stored in a "prescriptions" or "medication\_orders" table. Interactions between addresses and devices might be in a "device\_shipment" table.  These record relationships by including the id numbers of the entities involved, along with other data like the date of the interaction and other details.  A table called "medication\_orders", for example, will include a field for the patient id, a field for the medication id, a field for the prescribing provider id, the date and time of the medication order, the dose and duration of the order, the mode of medication delivery, and other important details such as any notes made by the prescriber.
+
+Whether these tables of interactions have primary keys and their own identifiers depends, but they often will.  For example, a "device\_shipment" table might have a primary key called "device\_shipment\_id".
+
+### Why Bother With Normalization?
+
+To show why organizing data in a normalized way is helpful, let's use an example.
+
+Which of the following is more efficient to correct, if we want to change "orane juice" to "orange juice"? Is it easier to correct with the data stored in one table, or with the data in two tables, where items are given an id number?
+
+**One Table Option**
+
+**orders**
+
+<!-- data-type="none" class="tight-table" -->
+| order_num   | item_1   | item_2  | ...  | item_50  |
+| :--------- | :--------- | :--------- | :--------- | :--------- |
+| 23125    | orane juice     | pistachios     | ...    | `NULL`    |
+| 41320    | peanut butter    | plain bagels     | ...    | orane juice    |
+| 53011    | napkins    | distilled water     | ...    | `NULL`    |
+| 14123    | pistachios    | orane juice     | ...    | peanut butter    |
+
+In the one table option, we have to look for the phrase "orane juice" anywhere in 50 columns across all rows (consider that there could be thousands of rows).
+
+**Two Table Option**
+
+**items**
+
+*Note: here we are using the "item\_id" field as a primary key as well, which is why we include (PK) in the table header.  Often, database tools will do something similar and display a special symbol (like <span class="fa-solid fa-key"></span>) or the letters "PK" to draw your attention to the fact that a particular field is a primary key.* 
```

Comment:
Maybe combine this note and the note under order\_items into a single "important" highlight box?

Reply:
I like it, that's a good use of that box!